backend/src/v2/driver/driver.go

```diff
@@ -665,6 +665,33 @@ func extendPodSpecPatch(
 		podSpec.Volumes = append(podSpec.Volumes, ephemeralVolume)
 		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, ephemeralVolumeMount)
 	}
+
+	// EmptyDirMounts
+	for _, emptyDirVolumeSpec := range kubernetesExecutorConfig.GetEmptyDirMounts() {
+		var sizeLimitResource *k8sres.Quantity
```

Comment:
Could add a check for the podSpec.Volumes slice to see if a volume with the same name already exists. If a duplicate is found, it logs an error and skips the creation of the volume.
```
if emptyDirVolumeSpec.GetSizeLimit() != "" {
        r, err := k8sres.ParseQuantity(emptyDirVolumeSpec.GetSizeLimit())
        if err != nil {
            log.Errorf("Failed to parse size limit '%s' for volume '%s': %v",
                emptyDirVolumeSpec.GetSizeLimit(),
                emptyDirVolumeSpec.GetVolumeName(),
                err)
            continue // Skip this volume since size limit parsing failed
        }
        sizeLimitResource = &r
    }
    ```

Reply:
2 lines below we have
```
r := k8sres.MustParse(emptyDirVolumeSpec.GetSizeLimit())
```

`MustParse` will panic if the resource can't be parsed, and that makes sense to me. In other words, I don't think we should keep going and try to run the user's workload if the volume can't be mounted. If the volume can't be mounted, the workload is likely to fail anyway, so better to fail faster.